### PR TITLE
:bug: filter secret informer to reduce cache size

### PR DESCRIPTION
**Pull request by zhiweiyin318**
## Summary
the klusterlet watches the secrets and deployments. when there are large numbers of secrets  or deployments in the managed cluster, the pod of klusterlet will be OOM because the cache size of secret/deployment informer is increased beyond the limit. So this PR is to filter the secret and deployment informer to reduce the case size.

## Related issue(s)

Fixes #

### CODE REVIEW

**Comment by zhiweiyin318** [comment]
> we can reduce the permission, right?

@qiujian16  I am afraid not, the registration-agent also need secret permission to create addon hub-kubeconfig secrets which have different names.

File: pkg/operator/operators/klusterlet/options.go [hunk 1]
```diff
@@ -52,6 +56,24 @@ func (o *Options) RunKlusterletOperator(ctx context.Context, controllerContext *
 
 	kubeInformer := informers.NewSharedInformerFactory(kubeClient, 5*time.Minute)
 
+	// to reduce cache size if there are larges number of secrets
+	newOnTermInformer := func(name string) informers.SharedInformerFactory {
+		return informers.NewSharedInformerFactoryWithOptions(kubeClient, 5*time.Minute,
+			informers.WithTweakListOptions(func(options *metav1.ListOptions) {
+				options.FieldSelector = fields.OneTermEqualSelector("metadata.name", name).String()
```
Comment (zhujian7) [inline pkg/operator/operators/klusterlet/options.go:63]: Is there any expression of the selector to make it possible to be logic OR for the `metadata.name`?

File: pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_cleanup_controller.go [hunk 2]
```diff
@@ -62,7 +60,6 @@ func NewKlusterletCleanupController(
 	}
 
 	return factory.New().WithSync(controller.sync).
-		WithInformersQueueKeyFunc(helpers.KlusterletSecretQueueKeyFunc(controller.klusterletLister), secretInformer.Informer()).
```
Comment (zhujian7) [inline pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_cleanup_controller.go:65]: in hosted mode, we need the external-managed-kubecnfig to build a client to clean up resources on the managed cluster. so I think it is necessary to watch this secret to trigger a cleanup reconcile.

Comment (zhiweiyin318) [reply pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_cleanup_controller.go:65]: added back.

File: pkg/operator/operators/klusterlet/options.go (hunk 1 above)
Comment (zhiweiyin318) [reply pkg/operator/operators/klusterlet/options.go:63]: have not found that expression.

File: pkg/operator/operators/klusterlet/options.go [hunk 3]
```diff
@@ -52,6 +56,29 @@ func (o *Options) RunKlusterletOperator(ctx context.Context, controllerContext *
 
 	kubeInformer := informers.NewSharedInformerFactory(kubeClient, 5*time.Minute)
 
+	// to reduce cache size if there are larges number of secrets
+	newOnTermInformer := func(name string) informers.SharedInformerFactory {
```
Comment (zhujian7) [inline pkg/operator/operators/klusterlet/options.go:60]:
```suggestion
	newOneTermInformer := func(name string) informers.SharedInformerFactory {
```

Comment (zhiweiyin318) [reply pkg/operator/operators/klusterlet/options.go]: done

**Comment by zhujian7** [comment]
LGTM

**Pull request merged**